src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimal(animal) {
+    const index = Animal.alive.indexOf(animal);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      this.health = 0;
+      Animal.removeDeadAnimal(this);
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey.hidden || prey instanceof Carnivore || prey.health <= 0) {
```

Comment:
The condition `prey instanceof Carnivore` is redundant because the `bite` method should only be called with `Herbivore` instances. Consider removing this check and ensuring that the `bite` method is only used with `Herbivore` instances.